nilearn/reporting/html_report.py

```diff
@@ -0,0 +1,188 @@
+import io
+import base64
+import warnings
+from pathlib import Path
+from string import Template
+
+from nilearn.externals import tempita
+from nilearn.plotting import js_plotting_utils as plot_utils
+
+
+def _embed_img(display):
+    """
+    Parameters
+    ----------
+    display: obj
+        A matplotlib object to display
+
+    Returns
+    -------
+    embed : str
+        Binary image string
+    """
+    if display is None:  # no display, show single transparent pixel
+        data = ("iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAw" +
+                "CAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=")
+        return data
+
+    else:  # we were passed a matplotlib display
+        io_buffer = io.BytesIO()
+        display.savefig(io_buffer)
+        display.close()
+
+        io_buffer.seek(0)
+        data = base64.b64encode(io_buffer.read())
+
+        return '{}'.format(data.decode())
+
+
+def _str_params(params):
+    """
+    Convert NoneType values to the string 'None'
```

Comment:
could you add in the docstring that this modifies params inplace? or maybe you should modify and return a  copy?

Reply:
What does Nilearn generally do with other objects? In replace or copy? We should do that.

Reply:
This now returns a modified copy !